packages/components/src/tabs/stories/index.story.tsx

```diff
@@ -492,3 +497,125 @@ const TabGetsRemovedTemplate: StoryFn< typeof Tabs > = ( props ) => {
 	);
 };
 export const TabGetsRemoved = TabGetsRemovedTemplate.bind( {} );
+
+const WITH_LINKS_PAGES = [
+	{
+		title: 'Page 1',
+		url: 'page1',
+	},
+	{
+		title: 'Page 2',
+		url: 'page2',
+	},
+	{
+		title: 'Page 3',
+		url: 'page3',
+	},
+];
+
+const SimpleRouterContext = createContext< {
+	currentPath: string | undefined;
+	goTo: ( path: string | undefined ) => void;
+} >( {
+	currentPath: undefined,
+	goTo: () => {},
+} );
+
+const SimpleRouter = ( {
+	children,
+	initialRoute = '/',
+}: {
+	children: React.ReactNode;
+	initialRoute?: string;
+} ) => {
+	const [ currentPath, setCurrentPath ] = useState< string | undefined >(
+		initialRoute
+	);
+
+	const goTo = ( path: string | undefined ) => {
+		setCurrentPath( path );
+	};
+
+	return (
+		<SimpleRouterContext.Provider value={ { currentPath, goTo } }>
+			{ children }
+		</SimpleRouterContext.Provider>
+	);
+};
+
+const SimpleRouterLink = forwardRef(
+	(
+		{
+			to,
+			children,
+			onClick,
+			style,
+			...restProps
+		}: {
+			to: string;
+			children?: React.ReactNode;
+		} & React.HTMLProps< HTMLAnchorElement >,
+		ref: React.ForwardedRef< HTMLAnchorElement >
+	) => {
+		const { goTo } = useContext( SimpleRouterContext );
+
+		return (
+			<a
+				href={ to }
+				ref={ ref }
+				onClick={ ( event ) => {
+					event.preventDefault();
+					goTo( to );
+					onClick?.( event );
+				} }
+				style={ {
+					...style,
+					textDecoration: 'none',
+				} }
+				{ ...restProps }
+			>
+				{ children }
+			</a>
+		);
+	}
+);
+
+const WithLinksTabs = ( props: React.ComponentProps< typeof Tabs > ) => {
+	const { currentPath, goTo } = useContext( SimpleRouterContext );
+
+	return (
+		<Tabs
+			selectedTabId={ currentPath }
+			onSelect={ ( newTabId ) => {
+				goTo( newTabId ?? undefined );
+			} }
```

Comment:
This feels correct, even if it could be redundant since the `SimpleRouterLink` component also calls `goTo` internally. What do folks think:?